NSubstitute.DbConnection/NSubstitute.DbConnection/DbConnnectionMockExtensions.cs

```diff
@@ -145,6 +145,8 @@ private class MockQuery : IMockQueryBuilder, IMockQueryResultBuilder
 
             public List<(Type RowType, IReadOnlyList<object> Rows)> ResultSets { get; } = new List<(Type RowType, IReadOnlyList<object> Rows)>();
 
+            public Predicate<IDbCommand> Matcher { get; set; } = null;
```

Comment:
Predicate is a weird delegate type - it's not very widely used and also confuses people so I prefer Func<T, bool> over Predicate<T> just for "least surprise" purposes